sns/README.md

```diff
@@ -15,23 +15,288 @@ Current version is `0.1.0`.
+| <a name="channelBindingObjectFifo"></a>`type` | [ordering](#ordering)| **Optional.** By default, we assume an unordered SNS topic. This field allows configuration of a FIFO SNS Topic.  |
+| <a name="channelBindingObjectPolicy"></a>`policy` |[policy](#policy) | **Optional.** The security policy for the SNS Topic |
+| <a name="channelBindingObjectTags"></a>`tags` |Object | **Optional.** Key-value pairs that represent AWS tags on the topic. |
+|<a name="channelBindingObjectBindingVersion"></a>`bindingVersion` | string | **Optional**, defaults to `latest`. The version of this binding.|
+
+### Schemas
+
+#### Ordering
+|Field Name | Type | Description|
+|---|:---:|---|
+| <a name="channelBindingFifoObjectChannelType"></a>`type` | string one of: standard or FIFO | **Required.** What type of SNS Topic is this? |
+| <a name="channelBindingContentBasedDeduplication"></a>`contentBasedDepulication` | boolean | **Optional.** True to turn on de-duplication of messages for a channel.|
+
+#### Policy
+|Field Name | Type | Description|
+|---|:---:|---|
+| <a name="channelBindingPolicyObjectPolicyStatements"></a>`statements` | [Statement](#statement) | **Required.** An array of Statement objects, each of which controls a permission for this topic |
+
+#### Statement
+|Field Name | Type | Description|
+|---|:---:|---|
+| <a name="channelBindingPolicyStatementObjectEffect"></a>`effect` | string |**Required.** Either "Allow" or "Deny"|
+| <a name="channelBindingPolicyStatementObjectPrincipal"></a>`principal` | string or array of string |**Required.** The AWS account or IAM user to whom this statement applies|
+| <a name="channelBindingPolicyStatementObjectAction"></a>`action` | string or array of string |**Required.** The SNS permission being allowed or denied e.g. sns:Publish|
+
+##### Examples
 
+Just use defaults
 
+```yaml
+channels:
+  user-signedup:
+    description:  A user has signed up to our service
+    bindings:
+      sns: {} 
+```
+
+Minimal definition, just policy
+
+```yaml
+channels:
+  user-signedup:
+    description:  A user has signed up to our service
+    bindings:
+      sns:
+        policy:
+          statements: 
+          - effect : Allow
+            principal: *
+            action: SNS:Publish
+```
 
 <a name="operation"></a>
 
 ## Operation Binding Object
 
-This object MUST NOT contain any properties. Its name is reserved for future use.
+This object contains information operation binding in SNS.
+
+We represent SNS producers via a **subscribe** Operation Object. In simple cases this may not require configuration, and can be shown as an empty SNS Binding Object i.e. {} if you need to explicitly indicate how a producer publishes to the channel.
+
+We represent SNS consumers via a **publish** Operation Object. These consumers need an SNS Subscription that defines how they consume from SNS i.e. the protocol that they use, and any filters applied.
```

Comment:
Maybe I am not understanding something here. Why is an SNS consumer represented as a publish operation? This seems backwards to me. Shouldn't the SNS consumer use a subscribe operation?